src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -340,21 +351,49 @@ internal static void WhenPresent(this OpenApiObject apiObj, string str, Action<O
             }
         }
 
-        // See https://swagger.io/docs/specification/data-models/data-types/        
-        internal static ConnectorType ToConnectorType(this OpenApiParameter openApiParameter, ConnectorCompatibility compatibility, Stack<string> chain = null, int level = 0)
+        internal class ConnectorTypeGetterSettings
         {
-            chain ??= new Stack<string>();
+            internal readonly ConnectorCompatibility Compatibility;
+            internal Stack<string> Chain = new Stack<string>();
+            internal int Level = 0;            
+
+            internal ConnectorTypeGetterSettings(ConnectorCompatibility connectorCompatibility)
+            {
+                Compatibility = connectorCompatibility;
+            }          
 
+            internal ConnectorTypeGetterSettings Stack(string identifier)
+            {
+                Level++;
+                Chain.Push(identifier);
+                return this;
+            }
+
+            internal void UnStack()
+            { 
+                Chain.Pop();
+                Level--;
+            }
+        }
+
+        internal static ConnectorType GetConnectorType(this OpenApiParameter openApiParameter, ConnectorCompatibility compatibility)
+        {
+            return openApiParameter.GetConnectorType(new ConnectorTypeGetterSettings(compatibility));
+        }
+
+        // See https://swagger.io/docs/specification/data-models/data-types/        
+        internal static ConnectorType GetConnectorType(this OpenApiParameter openApiParameter, ConnectorTypeGetterSettings settings)
```

Comment:
>GetConnectorType [](http://example.com/codeflow?start=38&length=16)

Key method where exceptions have been removed
Errors are propagated in ConnectorType
ConnectorTypeGetterSettings class is shared between recursive calls